aten/src/ATen/templates/Functions.cpp

```diff
@@ -99,16 +103,17 @@ Tensor TensorMaker::make_tensor() {
 
   Storage storage{Storage::use_byte_size_t{}, size_bytes, std::move(data_ptr)};
 
-  at::Tensor tensor = makeEmptyTensor();
+  Tensor tensor = detail::make_tensor<TensorImpl>(
+      std::move(storage), opts_.computeDispatchKey(), opts_.dtype());
 
-  c10::TensorImpl* tensor_impl = tensor.unsafeGetTensorImpl();
+  if (!sizes_.empty() && sizes_[0] != 0) {
+    TensorImpl* tensor_impl = tensor.unsafeGetTensorImpl();
 
-  tensor_impl->set_storage_keep_dtype(std::move(storage));
-
-  if (strides_) {
-    tensor_impl->set_sizes_and_strides(sizes_, *strides_);
-  } else {
-    tensor_impl->set_sizes_contiguous(sizes_);
+    if (strides_) {
+      tensor_impl->set_sizes_and_strides(sizes_, *strides_);
+    } else {
+      tensor_impl->set_sizes_contiguous(sizes_);
+    }
```

Comment:
too bad you can't use `empty_generic` here (I think you probably could, but you'd need some refactoring)

Reply:
Well that was initially my intention, but later I realized the refactored overload of `empty_generic()` would end up with an implementation that is identical to what I have here. Considering that this overload would be solely used for constructing tensors from external data and that this is already the sole purpose of `TensorMaker::make_tensor()` I couldn't see the appeal of moving it to `empty_generic()`,